dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/openaire/OpenAireExportUtil.java

```diff
@@ -1577,4 +1540,68 @@ private static boolean isValidEmailAddress(String email) {
         }
         return result;
     }
+
+    private static void writeRightsUriLicenseInfoAttribute(XMLStreamWriter xmlw, List<FileDTO> files, String language) throws XMLStreamException {
+        if(!Lists.isEmpty(files)) {
+            writeRightsHeader(xmlw, language);
+
+            if(areAllFilesRestricted(files)) {
+                xmlw.writeCharacters("Access to all files in the dataset is restricted.");
+            } else if(areAllFilesAllRightsReserved(files)) {
+                xmlw.writeCharacters("All rights reserved.");
+            } else if(areAllFilesUnderSameLincese(files)) {
+                xmlw.writeAttribute("rightsURI", files.get(0).getLicenseUrl());
+                xmlw.writeCharacters(files.get(0).getLicenseName());
+            } else if(hasRestrictedFile(files)) {
+                xmlw.writeCharacters("Different licenses and/or terms apply to individual files in the dataset. Access to some files in the dataset is restricted.");
+            } else {
+                xmlw.writeCharacters("Different licenses and/or terms apply to individual files in the dataset.");
+            }
+
+            xmlw.writeEndElement();
+        }
+    }
+
+    private static boolean areAllFilesUnderSameLincese(List<FileDTO> files) {
+        final String firstFileLicense = files.get(0).getLicenseName();
+        if(StringUtils.isNotEmpty(firstFileLicense)) {
+            return files
+                    .stream()
+                    .allMatch(fileDTO -> fileDTO.getLicenseName().equals(firstFileLicense));
+        }
+        return false;
+    }
+
+    private static boolean areAllFilesAllRightsReserved(List<FileDTO> files) {
+        return files
+                .stream()
+                .allMatch(fileDTO -> fileDTO.getTermsOfUseType().equals(FileTermsOfUse.TermsOfUseType.ALL_RIGHTS_RESERVED.toString()));
+    }
+
+    private static boolean areAllFilesRestricted(List<FileDTO> files) {
```

Comment:
You can also create a method 
```
areAllFilesHasSameLicense(TermsOfUseType whatever) 
```
So you won't have to repeat the methods.

Reply:
The method would have to take FileDTO parameter aswell to make a comparison. There are literally 3 one-liners with `fileDTO.getTermsOfUseType().equals(FileTermsOfUse.TermsOfUseType.ALL_RIGHTS_RESERVED.toString())` - it is tempting to wrap those in a private method, but in thruth that would make code less readable in my opinion. 

Changing:
`fileDTO -> fileDTO.getTermsOfUseType().equals(FileTermsOfUse.TermsOfUseType.ALL_RIGHTS_RESERVED.toString())` to something like 
`fileDTO -> isOfTermsOfUseType(fileDTO, FileTermsOfUse.TermsOfUseType.ALL_RIGHTS_RESERVED)` isn't much of a gain IMO.